modules/porous_flow/include/materials/PorousFlowSingleComponentFluid.h

```diff
@@ -30,41 +30,50 @@ class PorousFlowSingleComponentFluid : public PorousFlowFluidPropertiesBase
   virtual void initQpStatefulProperties() override;
   virtual void computeQpProperties() override;
 
+  /// If true, this Material will compute density and viscosity, and their derivatives
+  const bool _compute_rho_nu;
+
+  /// If true, this Material will compute internal energy and its derivatives
+  const bool _compute_internal_energy;
+
+  /// If true, this Material will compute enthalpy and its derivatives
+  const bool _compute_enthalpy;
+
```

Comment:
I'm not sure. I was just thinking that they would always be needed together in a 'real' simulation - I didn't think of the steady-state case.

I have to add enthalpy and internal energy to the persistent variable materials (they are only isothermal at the moment), so I'll just use what you have here.